dartagnan/src/test/java/com/dat3m/dartagnan/AbstractDartagnanTest.java

```diff
@@ -64,19 +71,29 @@
         this.settings = settings;
     }
 
-    @Test
+    private void initSolverContext() throws Exception {
+        Configuration config = Configuration.builder()
+        		.setOption("solver.z3.usePhantomReferences", "true")
+        		.build();
+		ctx = SolverContextFactory.createSolverContext(
+                config, 
+                BasicLogManager.create(config), 
+                ShutdownManager.create().getNotifier(), 
+                Solvers.Z3);
+    }
+    
+    @Test()
     public void test() {
-    	try {
+        try {
             Program program = new ProgramParser().parse(new File(path));
-            if (program.getAss() != null) {
-                Context ctx = new Context();
-                Solver solver = ctx.mkSolver(ctx.mkTactic(Settings.TACTIC));
-                VerificationTask task = new VerificationTask(program, wmm, target, settings);
-                assertEquals(expected, runAnalysis(solver, ctx, task));
-                ctx.close();
-            }
+            VerificationTask task = new VerificationTask(program, wmm, target, settings);
+            initSolverContext();
```

Comment:
If this method returns the Context, a try-with-resource scope can be used to guarantee cleanup on case of exceptions.